d2l-hm-constants-behavior.html

```diff
@@ -68,7 +76,8 @@
 				userQuizActivity: 'user-quiz-activity'
 			},
 			assignments: {
-				assignment: 'assignment'
+				assignment: 'assignment',
+				instructions: 'instructions'
```

Comment:
Open question, but would we expect classes to only really have meaning when combined with a rel? i.e. does "this entity has the `instructions` class therefore it has `properties.instructions`" hold true, or would it be more "this entity has the `instructions` class _and also_ it has a rel of `https://api.brightspace.com/rels/assignment` therefore it has `properties.instructions`"?

I ask as I'm not sure if this should be `HypermediaClasses.instructions`, or `HypermediaClasses.assignments.instructions`; not sure!

Reply:
It's not a property though, but rather a sub-entity now. So you do something maybe like:

```
_getAssignmentInstructions: function(activity) {
			var instructions = activity.getSubEntityByRel('https://assignments.api.brightspace.com/rels/instructions');
			return instructions && instructions.hasClass('richtext') ? instructions.properties.text : '';
		}
```

Reply:
Er, yeah fair, although sort of the same question - this is more a question about what a class means in general, rather than what does this particular class mean. i.e. which of the following makes more sense to you:

1. `foobar` ∈ entity.class ∴ entity has Constraint A; or
2. `foobar` ∈ entity.class AND `barbaz` ∈ entity.rel ∴ entity has Constraint A

where Constraint A could be a property, the presence of a link/action/subentity, or anything else really.

The "actual question" behind this is really just asking whether this should be `HypermediaClasses.assignments.instructions`, or just `HypermediaClasses.instructions` - as in, does having a class of `instructions` mean different things in different places. I'm thinking what you've got makes the most sense (i.e. statement 2 above is more correct), but wanted to make sure!

Reply:
Yeah that was my thinking as well, that we have a very general concept of a `richtext` thing, and also a somewhat general concept of an `instructions` thing, and also a general concept of a `description` thing. We don't expose/use them, but I think quizzes have instructions, too. So theoretically a UI somewhere might be like 'for all of these activity things, if they have `instructions` on them, I'll grab those and assuming they are `richtext` like I expect I can format 'em and throw 'em on this page here and it will logically make sense. Or maybe one day we have some that get written out as `supertext` so we create a renderer for that, but we still know that logically they are all `instructions` regardless of whether they are `richtext` or `supertext` and so we can collect 'em all together on some screen or widget or whatever.

Anyways, I think that's the idea, and that's why I kept them out of the `assignments` namespace here.